\hypertarget{classjetfuel_1_1core_1_1exceptions_1_1Out__of__vector__range__exception}{}\section{jetfuel\+:\+:core\+:\+:exceptions\+:\+:Out\+\_\+of\+\_\+vector\+\_\+range\+\_\+exception Class Reference}
\label{classjetfuel_1_1core_1_1exceptions_1_1Out__of__vector__range__exception}\index{jetfuel\+::core\+::exceptions\+::\+Out\+\_\+of\+\_\+vector\+\_\+range\+\_\+exception@{jetfuel\+::core\+::exceptions\+::\+Out\+\_\+of\+\_\+vector\+\_\+range\+\_\+exception}}


{\ttfamily \#include $<$messagebus.\+h$>$}



Inheritance diagram for jetfuel\+:\+:core\+:\+:exceptions\+:\+:Out\+\_\+of\+\_\+vector\+\_\+range\+\_\+exception\+:\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=241pt]{classjetfuel_1_1core_1_1exceptions_1_1Out__of__vector__range__exception__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for jetfuel\+:\+:core\+:\+:exceptions\+:\+:Out\+\_\+of\+\_\+vector\+\_\+range\+\_\+exception\+:\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=241pt]{classjetfuel_1_1core_1_1exceptions_1_1Out__of__vector__range__exception__coll__graph}
\end{center}
\end{figure}


\subsection{Detailed Description}
This exception is thrown when a function requests for data from a vector location that is outside of that vector\textquotesingle{}s range.

Code Example\+: 
\begin{DoxyCode}
\textcolor{keywordtype}{int} Get\_int\_in\_vector(\textcolor{keyword}{const} \textcolor{keywordtype}{int} location)\textcolor{keyword}{ const }\{
    \textcolor{keywordflow}{if}(intvector.size() <= location)\{
        \textcolor{keywordflow}{throw} exceptions::Out\_of\_vector\_range\_exception();
    \}\textcolor{keywordflow}{else}\{
        \textcolor{keywordflow}{return} intvector[location];
    \}
\}
\end{DoxyCode}
 

The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
include/jetfuelcore/messagebus.\+h\end{DoxyCompactItemize}
